### PR TITLE
Added Gui Python interface

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

This PR is part of this meta ticket https://github.com/ignitionrobotics/sdformat/issues/931

Require
 - https://github.com/ignitionrobotics/sdformat/pull/997

## Summary

Added Gui Python interface

## Test it

```python
from sdformat import Gui

gui = Gui()
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: python/src/sdf/pyGui.cc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  pybind11::class_<sdf::Gui>(module, "Gui")
+    .def(pybind11::init<>())
+    .def(pybind11::init<sdf::Gui>())
+    .def(pybind11::self == pybind11::self)
+    .def("fullscreen", &sdf::Gui::Fullscreen,
+         "Get whether the Gui should be fullscreen.")
+    .def("set_fullscreen", &sdf::Gui::SetFullscreen,
+         "Set whether the Gui should be full screen.")
+    .def("plugin_count", &sdf::Gui::PluginCount,
+         "Get the filename of the shared library.")
```
Comment (scpeters) [inline python/src/sdf/pyGui.cc:45]: "Get the number of plugins."

**Review by azeey — CHANGES_REQUESTED** [review]

File: python/test/pyWorld_TEST.py [hunk 2]
```diff
@@ -14,13 +14,13 @@
...
 from ignition.math import Pose3d, Vector3d, SphericalCoordinates
-from sdformat import Plugin, Error, Frame, Light, Model, World
+from sdformat import Atmosphere, Gui, Plugin, Error, Frame, Light, Model, World
 import sdformat as sdf
 import unittest
 import math
 
 # TODO(ahcorde)
-# - Add Atmosphere, GUI, Actor, Light, Scene and Plugin tests when the sdf::Classes are ported
+# - Add Actor, Physics and sSene tests when the sdf::Classes are ported
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply python/test/pyWorld_TEST.py]: https://github.com/gazebosim/sdformat/pull/998/commits/d86e474af222c27e02b95c3f09729eb9a6d21ae5

File: python/test/pyGui_TEST.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+from sdformat import Gui, Plugin
+import unittest
+
+
+class PluginTEST(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply python/test/pyGui_TEST.py]: https://github.com/gazebosim/sdformat/pull/998/commits/d86e474af222c27e02b95c3f09729eb9a6d21ae5

File: python/src/sdf/pyGui.cc [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    .def("plugin_by_index", &sdf::Gui::PluginByIndex,
+         pybind11::return_value_policy::reference,
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply python/src/sdf/pyGui.cc]: https://github.com/gazebosim/sdformat/pull/998/commits/d86e474af222c27e02b95c3f09729eb9a6d21ae5

File: python/src/sdf/pyGui.cc [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+         "Get a plugin based on an index.")
+    .def("clear_plugins", &sdf::Gui::ClearPlugins,
+         "Remove all plugins")
+    .def("add_plugin", &sdf::Gui::AddPlugin,
+         "Add a plugin to this object.")
+    .def("plugins",
+         pybind11::overload_cast<>(&sdf::Gui::Plugins),
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply python/src/sdf/pyGui.cc:54]: https://github.com/gazebosim/sdformat/pull/998/commits/d86e474af222c27e02b95c3f09729eb9a6d21ae5

File: python/src/sdf/pyGui.cc (hunk 1 above)
Comment (ahcorde) [reply python/src/sdf/pyGui.cc]: https://github.com/gazebosim/sdformat/pull/998/commits/d86e474af222c27e02b95c3f09729eb9a6d21ae5

**Review by azeey — APPROVED** [review]

**Pull request merged**